R/utils_general.R

```diff
@@ -14,3 +14,47 @@
            width = 6,
            scientific = FALSE)
 }
+
+#' @title Update documentation
+#' 
+#' @description This functions generates the documentation for a package if both /man and /documentation folders exist within the project:
+#'   \itemize{
+#'     \item{devtools::document(pkg=pkg) is used to generate Rd files}
+#'     \item{Rd files are rendered using ?'anyfunction' eg ?visR}
+#'     \item{Documentation is collected within 'pkg'.pdf. This pdf is written to the documentation folder}
+#'     }
+#'
+#' @usage update_documentation(pkg = "~/visR")
+#' 
+#' @param pkg Path to package.
+#'
+#' @return .Rd files inside man/ folder and an updated pdf within the documentation folder.
+#' @noRd
+#' 
+#' @examples
+#' \dontrun{
+#'   update_documentation(pkg = "~/visR")
+#'  }
+
+update_documentation <- function(pkg = "~/visR", fileNm="visR")
+  {
+
+    ## Create Rd files document ####
+    devtools::document(pkg = pkg)
```

Comment:
FYI, this adds a {devtools} dependency (a pretty big one). If you choose to keep it, it needs to be added to DESCRIPTION

Reply:
Also if it is only for internal usage?

Reply:
> 1. You added a `documentation` folder. If you keep this folder, it needs to be added to `.Rbuildignore` to avoid a NOTE on r CMD Check. NOTES on R CMD Check will exclude the package from auto-acceptance on CRAN, and will kick it to a human for review.
Thanks for the note. I did not realize it.

> 2. FYI R 4.2 was released yesterday, and they have added an additional check for using `inherits(strata, "list")` instead of `class(strata) == "list"`, for example. Perhaps not in this PR, but all these calls will need to be updated. While you're updating, note that `inherits(x, c("a", "b"))` is equivalent to `inherits(x, "a") | inherits(x, "b")`.
Will you make an issue for this? 

> 3. For generic functions like `visr()`, where the the generic and S3 methods are all documented in a single file–please move all documentation of arguments to the top of the file. This makes it much easier to review, reduces risk of argument doc duplication, and adds parity between the documentation code and the rendered help file (where all documented arguments all appear on top).
Agree and updated

Reply:
> Also if it is only for internal usage? I have moved it to my local folder now.

Yes, because the code was in the R folder, and was shipped as part of the package, it would need to be listed as a dependency.